Running FedStream with: Dir=1.0, Init_Class=8, Epoch=10
Files already downloaded and verified
Files already downloaded and verified
has read
Files already downloaded and verified
Files already downloaded and verified
  0%|          | 0/100 [00:00<?, ?it/s]  1%|          | 1/100 [04:24<7:16:01, 264.26s/it]  1%|          | 1/100 [06:09<10:08:51, 369.01s/it]
Traceback (most recent call last):
  File "/home/kemove/lt/FedAlgorithm/alg/FedStream.py", line 775, in <module>
    alg.online_train()
  File "/home/kemove/lt/FedAlgorithm/alg/FedStream.py", line 738, in online_train
    global_loss_list, accuracy_list = self.alg.run(
                                      ^^^^^^^^^^^^^
  File "/home/kemove/lt/FedAlgorithm/alg/FedProx.py", line 113, in run
    result = self.client_group.clients[k].local_update_prox(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/lt/FedAlgorithm/alg/Clients.py", line 235, in local_update_prox
    for batch in dataloader:
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 631, in __next__
    data = self._next_data()
           ^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torch/utils/data/dataloader.py", line 675, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
            ~~~~~~~~~~~~^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torch/utils/data/dataset.py", line 348, in __getitem__
    return self.datasets[dataset_idx][sample_idx]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/home/kemove/lt/FedAlgorithm/alg/Clients.py", line 37, in __getitem__
    data = self.trans(self.local_data[index])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torchvision/transforms/transforms.py", line 95, in __call__
    img = t(img)
          ^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torchvision/transforms/transforms.py", line 234, in __call__
    return F.to_pil_image(pic, self.mode)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/torchvision/transforms/functional.py", line 324, in to_pil_image
    return Image.fromarray(npimg, mode=mode)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/kemove/anaconda3/envs/test/lib/python3.11/site-packages/PIL/Image.py", line 3150, in fromarray
    obj = obj.tobytes()
          ^^^^^^^^^^^^^
KeyboardInterrupt
